evalml/pipelines/components/estimators/regressors/varmax_regressor.py

```diff
@@ -0,0 +1,243 @@
+        self,
+        time_index: Optional[Hashable] = None,
+        p: int = 1,
+        q: int = 0,
+        trend: Optional[str] = "c",
+        random_seed: Union[int, float] = 0,
+        maxiter: int = 10,
+        use_covariates: bool = True,
+        **kwargs,
+    ):
+        self.preds_95_upper = None
+        self.preds_95_lower = None
+        parameters = {
+            "order": (p, q),
+            "trend": trend,
+            "maxiter": maxiter,
+        }
+        parameters.update(kwargs)
+
+        varmax_model_msg = (
+            "sktime is not installed. Please install using `pip install sktime.`"
+        )
+        sktime_varmax = import_or_raise(
+            "sktime.forecasting.varmax",
+            error_msg=varmax_model_msg,
+        )
+        varmax_model = sktime_varmax.VARMAX(**parameters)
+
+        parameters["use_covariates"] = use_covariates
+        parameters["time_index"] = time_index
+
+        self.use_covariates = use_covariates
+        self.time_index = time_index
+
+        super().__init__(
+            parameters=parameters,
+            component_obj=varmax_model,
+            random_seed=random_seed,
+        )
+
+    def _set_forecast_horizon(self, X: pd.DataFrame):
+        from sktime.forecasting.base import ForecastingHorizon
+
+        # we can only calculate the difference if the indices are of the same type
+        units_diff = 1
+        if isinstance(X.index[0], type(self.last_X_index)):
+            if isinstance(
+                X.index,
+                pd.DatetimeIndex,
+            ):
+                dates_diff = pd.date_range(
+                    start=self.last_X_index,
+                    end=X.index[0],
+                    freq=X.index.freq,
+                )
+                units_diff = len(dates_diff) - 1
+            elif X.index.is_numeric():
+                units_diff = X.index[0] - self.last_X_index
+        fh_ = ForecastingHorizon(
+            [units_diff + i for i in range(len(X))],
+            is_relative=True,
+        )
+        return fh_
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.DataFrame] = None):
+        """Fits VARMAX regressor to data.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features].
+            y (pd.DataFrane): The target training data of shape [n_samples, n_series_id_values].
+
+        Returns:
+            self
+
+        Raises:
+            ValueError: If y was not passed in.
+        """
+        X, y = self._manage_woodwork(X, y)
+
+        if y is None:
+            raise ValueError("VARMAX Regressor requires y as input.")
+
+        if X is not None and self.use_covariates:
+            self.last_X_index = X.index[-1]
+            X = X.ww.select(exclude=["Datetime"])
+
+            X.ww.set_types(
+                {
+                    col: "Double"
+                    for col in X.ww.select(["Boolean"], return_schema=True).columns
+                },
+            )
```

Comment:
This block shows up a couple times - can we consolidate into a method?

Reply:
Sure, consolidated into `convert_bool_to_double()`